### ISSUE TITLE
disabled TV-Channels are active again after a reboot

### ISSUE THREAD

**Issue by eldersixpack13** [xbmc/xbmc#25571]
### Describe the bug

disabled TV-Channels are active again after a reboot
it's a regression since the new database version (TV45.db) was introduced ( since Piers ???).
First saw new DB's some days ago.

## Expected Behavior

disabled channels should stay disabled

## Actual Behavior
disabled channels are enabled again after reboot

## Possible Fix
 no idea

### To Reproduce
1.  disable a TV channel under "PVR & Live TV" (see attached photo)
2. take a look into EPG: channel is gone as indicated
3. reboot
4. take a look into EPG: channel is there again
5. take a look under "PVR & Live TV" (see attached photo): the channel is active again

### Debuglog
none

### Screenshots 
![ChannelManager](https://github.com/user-attachments/assets/d15de7bd-8afe-435e-8077-56b645e8e973)

### Your Environment
Used Operating system:

 - [x ] Linux
 
 - Operating system version/name: LE13, nightly (20240804-a395fe7)
 - Kodi version, see: https://github.com/LibreELEC/LibreELEC.tv/pull/9147/files
 - Hardware Platform: NUC13
 - it's was a complete fresh/clean LE13 installation ~2 weeks ago, after I saw this bug first

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by ksooo** [comment]
> reboot

How exactly do you reboot? By pulling the power plug off the device? Via Kodi menu item? ...

If Kodi gets shut down properly, all data will be saved, including channel data. I just checked that latest Kodi master behaves properly. 

Without a full Kodi debug log for both the shutdown and startup after that I cannot help.

**Comment by eldersixpack13** [comment]
.kodi/temp/ was emtied after Debug was set to ON !

in my opinion you already have the proper log's ! 

AFAIK when kodi/LE starts it copies the elder kodi.log to kodi.old.log and generates a new kodi.log
Right ?
if so, then
kodi.old.log is the log of the previous run (box initial came up with WDR enabled and during this kodi run the channel was disabled). 
Then a reboot via GUI took place and and the kodi.log *is* the one from the last run, where WDR should stay disabled (it isn't !)

**Comment by eldersixpack13** [comment]
[DBs.zip](https://github.com/user-attachments/files/16499587/DBs.zip)

zip contains 4x DB's (naming should tell what I've done; with running numbers: 1_ to 4_ ) and 1x MD (md5sum's)

for me (looking at the mdsum's only) the DB (with the disabled channels) gets altered again, when kodi restarts

**Comment by ksooo** [comment]
Can you please enable PVR in component debug logging settings and send kodi.log from the kodi start after disabling the channel. for some reason, on startup, the channel in question gets deleted from the database, then recreated. The former deletes the information from the database that you disabled the channel locally.

**Comment by eldersixpack13** [comment]
for  both *.log's PVR and DB component logging enable, Kodi debug on

kodi.old.log (after reboot and afterwards disabling channel "WDR HD Köln")
[kodi.old.log](https://github.com/user-attachments/files/16500657/kodi.old.log)

kodi.log (after systemctl restart kodi.service)

[kodi.log](https://github.com/user-attachments/files/16500669/kodi.log)

**Comment by ksooo** [comment]
Are you able to apply a patch and to re-compile?

**Comment by eldersixpack13** [comment]
unfortunately no

**Comment by ksooo** [comment]
That’s bad. I have an idea would could be wrong, but cannot reproduce the issue myself.

**Comment by ksooo** [comment]
Sorry, I have no build environment for LibreELEC. All I can provide is a patch file. I'm pretty sure it fixes the problem but cannot test myself.

```diff
diff --git a/xbmc/pvr/channels/PVRChannelGroup.cpp b/xbmc/pvr/channels/PVRChannelGroup.cpp
index a7339ed1a915d..f802ba1a7ea40 100644
--- a/xbmc/pvr/channels/PVRChannelGroup.cpp
+++ b/xbmc/pvr/channels/PVRChannelGroup.cpp
@@ -783,9 +783,12 @@ bool CPVRChannelGroup::RemoveFromGroup(
     }
   }
 
-  // no need to renumber if nothing was removed
+  // no need to delete and renumber if nothing was removed
   if (bReturn)
+  {
+    DeleteGroupMembersFromDb({std::make_shared<CPVRChannelGroupMember>(*groupMember)});
     Renumber();
+  }
 
   return bReturn;
 }
@@ -878,11 +881,6 @@ void CPVRChannelGroup::Delete()
   }
 }
 
-void CPVRChannelGroup::DeleteGroupMember(const std::shared_ptr<CPVRChannelGroupMember>& member)
-{
-  DeleteGroupMembersFromDb({member});
-}
-
 bool CPVRChannelGroup::Renumber(RenumberMode mode /* = NORMAL */)
 {
   bool bReturn(false);
diff --git a/xbmc/pvr/channels/PVRChannelGroup.h b/xbmc/pvr/channels/PVRChannelGroup.h
index 0ce1bed4faaa4..f75cb00b5eb3a 100644
--- a/xbmc/pvr/channels/PVRChannelGroup.h
+++ b/xbmc/pvr/channels/PVRChannelGroup.h
@@ -441,12 +441,6 @@ class CPVRChannelGroup : public IChannelGroupSettingsCallback
    */
   void Delete();
 
-  /*!
-   * @brief Remove the given group member from the database.
-   * @param member The member to remove from the database.
-   */
-  void DeleteGroupMember(const std::shared_ptr<CPVRChannelGroupMember>& member);
-
   /*!
    * @brief Whether this group is deleted.
    * @return True, if deleted, false otherwise.
diff --git a/xbmc/pvr/channels/PVRChannelGroups.cpp b/xbmc/pvr/channels/PVRChannelGroups.cpp
index d15c8c530f081..55af6dd8da4f9 100644
--- a/xbmc/pvr/channels/PVRChannelGroups.cpp
+++ b/xbmc/pvr/channels/PVRChannelGroups.cpp
@@ -764,8 +764,6 @@ bool CPVRChannelGroups::RemoveFromGroup(const std::shared_ptr<CPVRChannelGroup>&
 
     if (group->RemoveFromGroup(groupMember))
     {
-      group->DeleteGroupMember(groupMember);
-
       // Changes in the all channels group may require resorting/renumbering of other groups.
       if (group->IsChannelsOwner())
         UpdateChannelNumbersFromAllChannelsGroup();
```

**Comment by eldersixpack13** [comment]
@ksooo thanks for support.
I'll test and report when it lands in LE13

**Comment by eldersixpack13** [comment]
confirming the patch, landed in LE13 nightly-20240820-a25491a, fixed the bug

thanks

**Comment by ksooo** [comment]
> confirming the patch, landed in LE13 nightly-20240820-a25491a, fixed the bug

Thanks for confirming the fix.
